Modules/OptionHolder.cs

```diff
@@ -3821,6 +3847,43 @@ public static void SetupRoleOptions(int id, TabGroup tab, CustomRoles role, Cust
         CustomRoleSpawnChances.Add(role, spawnOption);
         CustomRoleCounts.Add(role, countOption);
     }
+    public static void SetupGhostRoleOptions(int id, TabGroup tab, CustomRoles role, CustomGameMode customGameMode = CustomGameMode.Standard, bool zeroOne = false)
+    {
+        var spawnOption = StringOptionItem.Create(id, role.ToString(), ratesZeroOne, 0, tab, false).SetColor(Utils.GetRoleColor(role))
+            .SetHeader(true)
+            .SetGameMode(customGameMode) as StringOptionItem;
+
+        var countOption = IntegerOptionItem.Create(id + 1, "Maximum", new(1, 15, 1), 1, tab, false)
+         .SetParent(spawnOption)
+            .SetValueFormat(OptionFormat.Players)
+            .SetGameMode(customGameMode);
+
+        trueghostenable = false;
+        if(CustomGhostRoleCounts.ContainsKey(role)) // Just incase
+            CustomGhostRoleCounts.Remove(role);
+        
+
+        CustomRoleSpawnChances.Add(role, spawnOption);
+        CustomRoleCounts.Add(role, countOption);
+        GetRoleSpawnMode(role);
+        
+        if (trueghostenable) 
```

Comment:
Need help here, this dosen't work for some reason... I need help fidning a way to check if the setting is (0 or 1) before it get sent to all the other places (to prevent ghost-role from spawning)

Reply:
refer to "GetRoleSpawnMode" method.

Reply:
nvm, I think Ik the problem, it only runs after game starts I think

Reply:
okay... so ongamestart patch I go trough a foreach (var plrrole in customrolescounts) which insttantly leaves if not ghost role. And this finally made it possible to detect on/off option.

LMK if you have any better solution (PS: doing almost anything in optionholder fucks shit up. and RandomSpawnPatch is run before Load() so I can't use setupghostroleoptions)

Reply:
Sorry but why do you need a "setupghostroleoptions"

Reply:
> Sorry but why do you need a "setupghostroleoptions"

To be able and add them to the tab, and disable them from spawning normally, ghost roles will only spawn after a player has died. (Similar to setupadtroles, but I coulnd't quite find out how that disables spawing. So I found another method to do it)

Reply:
wait... I'm gonna look at how the code looks now, you maybe right and it's uneccessaary, but I think not because I wouldn't use setuadtroleoption, but I'll see...